proposed/http-client/http-client.md

```diff
@@ -148,7 +147,8 @@ use Psr\Http\Client\Exception;
  * Thrown when a response was received but the request itself failed.
  *
  * This exception MAY be thrown on HTTP response codes 4xx and 5xx.
- * This exception MUST NOT be thrown when using the client's default configuration.
+ * A comsuming library SHOULD NOT except this exception to be thrown. The
+ * exception exists for middlewares, third party library and application code.
```

Comment:
This doesn't make any sense to me. Either it can be thrown, then consumers should expect it, or it can't be, then it can be deleted.

Reply:
=> #41